bin/journey-test.sh

```diff
@@ -233,15 +233,15 @@ solve_all_exercises() {
 
     pushd ${exercism_exercises_dir}/${TRACK}/${exercise}
     # Check that tests compile before we strip @Ignore annotations
-    "$EXECPATH"/gradlew compileTestJava
+    "$EXECPATH"/gradlew compileTestJava --configure-on-demand --parallel
```

Comment:
Configuration on demand is incubating when used alongside parallel runs, according to the build output. Can you explain a little about why it might be expected to be robust here?